### DISCUSSION TITLE
the procedure to update the ccpp-physics in ccpp-scm

### DISCUSSION THREAD

**Discussion by HelinWei-NOAA** [NCAR/ccpp-scm#250] [General]
I would like to use the latest version of Noah MP in ccpp-scm. Below is what I did:

(1) git clone --recursive -b v5.0.0 https://github .com/NCAR/ccpp-scm
(2) point to the master branch of ccpp-physics and ccpp-framework
cd ccpp-scm/ccpp/physics
git checkout master
git pull
cd ../framework
git checkout master
git pull
(3)edit ccpp-scm/ccpp/config/ccpp_prebuild_config.py to reflect the name change of Noah MP driver (from sfc_noahmp_drv.f to sfc_noahmp_drv.F90)
(4) replace GFS_typedefs.F90 and GFS_typedefs.meta in ccpp-scm/scm/src by those in FV3/ccpp/data. I know I need to make some changes to both of them to get them working with the whole SCM.

Did I miss any other changes I need to make?

Thanks.

**Comment by grantfirl** [comment]
@HelinWei-NOAA 

If you want to use the latest version of Noah MP that is in ccpp-physics, you will not be able to use the CCPP-SCM without quite a bit of work. The v5 code of ccpp-physics was cut from the main development branch back in July/August of 2020 and the SCM v5 code is configured to work with it, so it also is fairly old.

If you want to use the latest ccpp-physics development code with the latest NoahMP scheme, you can use the latest SCM code which is up-to-date with the ccpp-physics.

If you do

git clone --recursive -b master https://github.com/NCAR/ccpp-scm

that should get you the latest code and I just tested that it works with the latest NoahMP code in ccpp-physics.

**Comment by HelinWei-NOAA** [comment]
What is the difference between the latest version of Noah MP that is in ccpp-physics and the one in  the latest ccpp-physics development code? The former is in the master of NCAR/ccpp-physics and which branch of ccpp-physics is used in the CCPP-SCM V5?

**Reply by grantfirl** [reply]
The "latest version of Noah MP" that I'm referring to is the one in the development code ("master" branch) of ccpp-physics. For the CCPP-SCM v5, the release/public-v5 branch of ccpp-physics is used. You can compare the branches on GitHub to see the differences, but I know that the release/public-v5 branch does not include the changes that you and Mike Barlage put in in late February.

**Reply by HelinWei-NOAA** [reply]
git clone --recursive -b master https://github.com/NCAR/ccpp-scm 
this indeed give me the  latest NoahMP code in ccpp-physics and the default compiling is working. I am going to create some suites using Noah MP to check if it is still working. Thanks.
